gax-java/gax/src/main/java/com/google/api/gax/rpc/EndpointContext.java

```diff
@@ -196,6 +203,10 @@ public abstract static class Builder {
      */
     public abstract Builder setTransportChannelProviderEndpoint(String transportChannelEndpoint);
 
+    public abstract Builder setUseS2A(boolean useS2A);
+
+    public abstract Builder setEnvProvider(EnvironmentProvider envProvider);
```

Comment:
Can we make this package-private if this is only being used in tests?

Reply:
Done in https://github.com/googleapis/sdk-platform-java/pull/3326/commits/9901656a2714d58cc6b2c0775da1bfdd70a4ddec